src/components/lib/sl/sl_raw.c

```diff
@@ -0,0 +1,391 @@
+/**
```

Comment:
wasn't a lot of this already in the code?  Did you not do a `git mv`?  If so, then we're losing all of the history on this...

Reply:
Omg. I have no idea how this got lost. 
I thought I did `git mv sl.c sl_sched.c` and then moved out the allocations into new files. But I see that I messed up somewhere and sl_sched.c doesn't have history. I'll try to see if I can recover the history in anyway now. I'm sorry!